earthvision/datasets/xview.py

```diff
@@ -1,2 +1,188 @@
+        # if exist, load and return
+        if coords_exists and chips_exists and classes_exists:
+            coords = np.load(coords_path)
+            chips = np.load(chips_path)
+            classes = np.load(classes_path)
+            return coords, chips, classes
+        
+        # read xView_train.geojson
+        fname = os.path.join(self.root, 'xView_train.geojson')
+        with open(fname) as f:
+            data = json.load(f)
+
+        # initialize
+        coords = []
+        chips = []
+        classes = []
+
+        # extract
+        feat_len = len(data['features'])
+        img_files = os.listdir(os.path.join(self.root, self.resources[0].split('.')[0]))
+
+        for i in range(feat_len):
+            properties = data['features'][i]['properties']
+            b_id = properties['image_id']
+            val = [int(num) for num in properties['bounds_imcoords'].split(',')]
+
+            # type_id 75 and 82 don't belong to any class
+            # https://github.com/DIUx-xView/xView1_baseline/issues/3
+            if properties['type_id'] not in [75, 82] and b_id in img_files:
+                chips.append(b_id)
+                classes.append(properties['type_id'])
+                coords.append(val)
+        
+        # convert to numpy arrays and save
+        coords = np.array(coords)
+        chips = np.array(chips)
+        classes = np.array(classes)
+        np.save(coords_path, coords)
+        np.save(chips_path, chips)
+        np.save(classes_path, classes)
+        
+        return coords, chips, classes
+    
+    def __getitem__(self, idx):
+        """Returns a tensor image and a dictionary of target 
+        consists of bounding boxes and labels
+        """
+        if self.data_mode == 'train':
+            # image
+            img_path = os.path.join(self.root, 'train_images', self.chips[idx])
+            image = _load_img(img_path)
+            image = np.array(image)
+            image = torch.from_numpy(image)
+
+            # bounding box
+            bbox = self.coords[idx]
+            bbox = torch.from_numpy(bbox)
+
+            # label
+            label = self.classes[idx]
+            label = np.vectorize(index_mapping.get)(label)
+            label = torch.from_numpy(label)
+            
+            # combine bounding box and label
+            target = {}
+            target['boxes'] = bbox
+            target['labels'] = label
+
+            sample = (image, target)
+        elif self.data_mode == 'validation':
+            # image
+            img_path = os.path.join(self.root, 'val_images', self.chips[idx])
+            image = _load_img(img_path)
+            image = np.array(image)
+            image = torch.from_numpy(image)
+
+            sample = image
+        else:
+            raise ValueError("data_mode not recognized. Try 'train' or 'validation'")
+        
+        return sample
+
+
+    def __len__(self):
+        return len(self.imgs)
+        
+
+    def __iter__(self):
```

Comment:
the function is no longer needed, we can do iter() without overriding magic method

Reply:
done